From 8416ba6a9b0e6d607655580b55c684cde6a435ee Mon Sep 17 00:00:00 2001
From: Ian Helmke <helmke@google.com>
Date: Fri, 10 Jun 2022 16:36:43 +0000
Subject: [PATCH 04/59] M-102: lacros - fix extension controlled pref observers

Currently creating an observer for an extension controlled pref causes
ash to crash. This is a prerequisite to the full implementation of
extension-controlled pref observers in lacros.

(cherry picked from commit c3fe354aeee8432539fe42ddb4b3ac3e71736cd4)

Bug: 1218145, 1334985
Change-Id: Iad18872f3614a4125c5623226a69c9848554512b
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3690828
Reviewed-by: Erik Chen <erikchen@chromium.org>
Commit-Queue: Ian Helmke <helmke@google.com>
Cr-Original-Commit-Position: refs/heads/main@{#1011757}
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3695789
Commit-Queue: Erik Chen <erikchen@chromium.org>
Reviewed-by: Hidehiko Abe <hidehiko@chromium.org>
Cr-Commit-Position: refs/branch-heads/5005@{#1163}
Cr-Branched-From: 5b4d9450fee01f821b6400e947b3839727643a71-refs/heads/main@{#992738}
---
 chrome/browser/ash/crosapi/prefs_ash.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/chrome/browser/ash/crosapi/prefs_ash.cc b/chrome/browser/ash/crosapi/prefs_ash.cc
index 5ddec3bda554..a9ade8d99ac1 100644
--- a/chrome/browser/ash/crosapi/prefs_ash.cc
+++ b/chrome/browser/ash/crosapi/prefs_ash.cc
@@ -302,7 +302,8 @@ absl::optional<PrefsAsh::State> PrefsAsh::GetState(mojom::PrefPath path) {
         return absl::nullopt;
       }
       std::string pref_name = GetExtensionPrefNameForPref(path);
-      return State{profile_prefs_registrar_->prefs(), nullptr, true, pref_name};
+      return State{profile_prefs_registrar_->prefs(),
+                   profile_prefs_registrar_.get(), true, pref_name};
     }
     default:
       LOG(WARNING) << "Unknown pref path: " << path;
-- 
2.37.0

